content/core-features/presence-and-occupancy/topic-presence_ia-update.textile

```diff
@@ -0,0 +1,737 @@
+channel.presence.enter("not moving", new CompletionListener() {
+  @Override
+  public void onSuccess() {
+    channel.presence.update("travelling North", new CompletionListener());
+  }
+});
+```
+
+```[csharp]
+/* Subscribe to presence enter and update events */
+channel.Presence.Subscribe(member =>
+{
+    switch (member.Action)
+    {
+        case PresenceAction.Enter:
+        case PresenceAction.Update:
+            {
+                Console.WriteLine(member.Data); // => travelling North
+                break;
+            }
+    }
+});
+
+/* Enter this client with data and update once entered */
+await channel.Presence.EnterAsync("not moving");
+await channel.Presence.UpdateAsync("travelling North");
+```
+
+```[ruby]
+# Subscribe to presence enter events
+channel.presence.subscribe(:enter) do |member|
+  puts member.data # => not moving
+end
+
+# Subscribe to presence update events
+channel.presence.subscribe(:update) do |member|
+  puts member.data # => travelling North
+end
+
+# Enter this client with data and update once entered
+channel.presence.enter(data: 'not moving') do
+  channel.presence.update(data: 'travelling North')
+end
+```
+
+```[objc]
+// Subscribe to presence enter events
+[channel.presence subscribe:ARTPresenceEnter callback:^(ARTPresenceMessage *member) {
+    NSLog(@"%@", member.data); // prints "not moving"
+}];
+
+// Subscribe to presence update events
+[channel.presence subscribe:ARTPresenceUpdate callback:^(ARTPresenceMessage *member) {
+    NSLog(@"%@", member.data); // prints "travelling North"
+}];
+
+// Enter this client with data and update once entered
+[channel.presence enter:@"not moving" callback:^(ARTErrorInfo *error) {
+    [channel.presence update:@"travelling North"];
+}];
+```
+
+```[swift]
+// Subscribe to presence enter events
+channel.presence.subscribe(.enter) { member in
+    print(member.data) // prints "not moving"
+}
+
+// Subscribe to presence update events
+channel.presence.subscribe(.update) { member in
+    print(member.data) // prints "travelling North"
+}
+
+// Enter this client with data and update once entered
+channel.presence.enter("not moving") { error in
+    channel.presence.update("travelling North")
+}
+```
+
+h4(#rest-member-data). REST API
+
+In addition to the <span lang="default">"@clientId@":/api/rest-sdk#client-options</span><span lang="ruby,python">"@client_id@":/api/rest-sdk#client-options</span><span lang="csharp">"@ClientId@":/api/rest-sdk#client-options</span> for members on a channel, it is also possible to include data when entering a channel. Clients can "update":/api/realtime-sdk/presence#update their data at any point which will be broadcasted to all presence subscribers as a <span lang="ruby">@:update@</span><span lang="java">@PresenceAction.UPDATE@</span><span lang="csharp">@PresenceAction.Update@</span><span lang="java">@Action.UPDATE@</span><span lang="default">@update@</span> event.
+
+
+h3(#member-list). Presence member list
```

Comment:
I would lose this h3 perhaps, and then more explicitly retitle the h4s. For example, "Realtime API" is too vague. It would need to be something like "Obtaining the presence list using the Realtime API", or whatever is approrpriate.